.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,40 @@
+name: Publish 📦 to PyPi 🚀
+
+on:
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+*"
+
+jobs:
+  build-n-publish:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.10.0
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.10.0
+
+    - name: Gets the tag version
+      id: context
+      run: |
+        echo ::set-output name=TAG_VERSION::${GITHUB_REF#refs/tags/}
+
+    - name: Setup the Python Environment by installing Poetry
+      uses: ./.github/actions/setup-python-build-env
+
+    - name: Code Quality Check
+      shell: bash
+      run: |
+        make install && make test
+
+    - name: Poetry bump version, build and publish
+      shell: bash
+      run: |
+        proj_version=$(poetry version -s)
+        if [ $proj_version != $TAG_VERSION ]; then echo "Version $proj_version, defined in pyproject.toml, does not match TAG $TAG_VERSION of this release"; exit 3; fi
+        poetry update
+        poetry publish --build -u __token__ -p $PYPI_TOKEN
```

Comment:
@OrangeTux , @Roger can you please setup the required secrets for this repo? I think just the PYPI_TOKEN is needed

Reply:
I don't have access to the project on Pypi.